binary_tree.c

```diff
@@ -128,6 +130,60 @@ static void tree_rotate_left(Tree* self, int value) {
     }
 }
 
+static void tree_rotate_right(Tree* self, int value) {
+
+    if(self->root->data == value && self->root->l_node) {
+        struct Node* new_subtree_root = self->root->l_node;
+
+        self->root->l_node = new_subtree_root->r_node;
+        new_subtree_root->r_node = self->root;
+        self->root = new_subtree_root;
+    }
+
+    else {
+        struct Node* parent = self->root;
+        struct Node* ex_subtree_root = NULL;
+        struct Node* new_subtree_root = NULL;
+
+        while(parent) {
+            if (parent->l_node && parent->l_node->data == value) {
+                ex_subtree_root = parent->l_node;
+
+                if(ex_subtree_root->l_node) {
+                    new_subtree_root = ex_subtree_root->l_node;
+
+                    parent->l_node = new_subtree_root;
+                    ex_subtree_root->l_node = new_subtree_root->r_node;
+                    new_subtree_root->r_node = ex_subtree_root;
+                }
+
+                break;
+            }
+            if (parent->r_node && parent->r_node->data == value) {
+                ex_subtree_root = parent->r_node;
+
+                if(ex_subtree_root->l_node) {
```

Comment:
this is a code duplication, you should avoid it `DRY` principle

Reply:
Here I agree, it can be expressed as an apart function for example
There is `WET` principle as well altough DRY indeed is better to apply here.